streams/src/main/java/org/apache/kafka/streams/state/internals/RecordConverter.java

```diff
@@ -47,5 +49,16 @@ static RecordConverter converter() {
                 record.leaderEpoch()
             );
         };
+
+        // privatize the constructor so the class cannot be instantiated (only used for its static members)
+        private RecordConverters() {}
+
+        public static RecordConverter rawValueToTimestampedValue() {
```

Comment:
Renamed the function to be a bit more explicit. It's internal, so the name shouldn't affect the public API at all.

Reply:
nit: `rawValueToTimestampedValueRecordConvert()` -- without `RecordConverter` suffix, it's unclear what it actually returns, a `RecordConverter`

Reply:
Are we doing Hungarian Notation now? https://en.wikipedia.org/wiki/Hungarian_notation ;)

Some might argue that it's unnecessary to encode the type in a name in a strongly (and explicitly) typed language like Java.

If you feel strongly about it, I'm fine with the change.

Reply:
For regular types I agree, however, here, we actually return a "function". I don't feel strong about it -- it's just personal taste I guess. If I read `RecordConverter.rawValueToTimestampedValue()` I would expect that `rawValueToTimestampedValue()` is a method that takes `rawValue` as input and returns a `timestampedValue` (as the method name indicates) -- but this is of course not the case (there is not even an input parameter) -- thus, it's confusing to me to read the code.

Reply:
I see.